packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -401,11 +401,19 @@ class TextSelectionOverlay {
   final DragStartBehavior dragStartBehavior;
 
   /// {@template flutter.widgets.TextSelectionOverlay.onSelectionHandleTapped}
-  /// A callback that's invoked when a selection handle is tapped.
-  ///
-  /// Both regular taps and long presses invoke this callback, but a drag
-  /// gesture won't.
+  /// A callback that's optionally invoked when a selection handle is tapped.
```

Comment:
```suggestion
  /// A callback that's optionally invoked based on platforms when a selection handle is tapped.
```

Reply:
I think ultimately this depends on what kind of selection control you're using, so it's not really platform-dependant. Even in `TextField`'s implementation you can override `textSelectionControl` so it's still up to the developer.

Anyways IMO we should deprecate this callback and let the `TextSelectionControl` implementation decides how the handles should be built and what interactions it should support. Currently it does not have enough information (as it doesn't have access to the `EditableTextState` that builds the selection overlay).